### DISCUSSION TITLE
Why clicking at button at once the newly entered text is not passed into modal component?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [livewire/livewire#8862] [1. Help]
On Laravel 11 / Livewire 3 site I have an input and button clicking on which modal is opened :

    <div class="flex max-h-10 p-0 m-0">
        <div class="p-0 w-full  m-0">
    
            <input id="text" name="text" type="text"
                   class="editor_form_input"
                   maxlength="255"
                   wire:model.live.change="noteText"
                   autocomplete="off"
                   placeholder="Fill string text "
            />
        </div>
        <div class="w-16 align-items-end justify-end align-top m-0 p-1">
            <button
                wire:click="$dispatch('openModal', { component: 'my-modal', arguments: {noteId: {{ $noteId }}, noteText: '{{ $noteText }}'} })">
                  Open
            </button>
    
        </div>
    </div>

$noteText - is public var defined in my component and opening modal I need to pass this var to modal.

It works OK if editing text I click out of input firstly (or click tab). But if I click at button at once, the newly entered text is not passed. How can I fix it ?

    "laravel/framework": "^11.9",
    "livewire/livewire": "^3.5",
    "wire-elements/modal": "^2.0"

**Comment by EdgarsJoja** [comment]
Do you need `$noteText` anywhere else as well? If not, I would consider using Alpine for input model binding. Sending server requests for something that is needed only in frontend is.... not ideal, makes everything less responsive than it needs to be.
```blade
<div x-data="{ noteText: '' }">
  <input (...) x-model="noteText" />
  <button @click="$dispatch(..., { ..., noteText })">Open</button>
</div>
```
something like that

**Reply by PetroGromovo** [reply]
the button has $dispatch defined - with modal opened.
In alpineJS $dispatch is a helpful shortcut for dispatching browser events.

Have I Dispatching to x-model ?
I try as :
```

        <button
            wire:click="$dispatch('openModal', { component: 'your-notebook-modal', arguments: {noteId: {{ $noteId }}, noteText: '{{ $noteText }}'} })" @click="$dispatch('input', noteText)">
        </button>

```

But I got error :

```
livewire.js?id=cc800bf4:1125 Alpine Expression Error: noteText is not defined

Expression: "$dispatch('input', noteText)"

```

As $noteText is defined in livewire component, not in alpinejs scope...

**Reply by EdgarsJoja** [reply]
that's why I asked if you need $noteText somewhere else (in BE). Because if not, then IMO it is much more efficient to define it only in alpine scope on FE, and use alpine instead of livewire logic.
Otherwise, you're making roundtrip to BE to retrieve value that you already had on FE.
